docs/data/base/components/slider/UnstyledSliderValueLabel.tsx

```diff
@@ -0,0 +1,132 @@
+
+const blue = {
+  100: '#DAECFF',
+  200: '#99CCF3',
+  400: '#3399FF',
+  300: '#66B2FF',
+  500: '#007FFF',
+  600: '#0072E5',
+  900: '#003A75',
+};
+
+const grey = {
+  50: '#f6f8fa',
+  100: '#eaeef2',
+  200: '#d0d7de',
+  300: '#afb8c1',
+  400: '#8c959f',
+  500: '#6e7781',
+  600: '#57606a',
+  700: '#424a53',
+  800: '#32383f',
+  900: '#24292f',
+};
+
+const StyledSlider = styled(SliderUnstyled)(
+  ({ theme }) => `
+  color: ${theme.palette.mode === 'light' ? blue[500] : blue[300]};
+  height: 6px;
+  width: 100%;
+  padding: 16px 0;
+  display: inline-block;
+  position: relative;
+  cursor: pointer;
+  touch-action: none;
+  -webkit-tap-highlight-color: transparent;
+  &:hover {
+    opacity: 1;
+  }
+  &.${sliderUnstyledClasses.disabled} { 
+    pointer-events: none;
+    cursor: default;
+    color: ${theme.palette.mode === 'light' ? grey[300] : grey[600]};
+    opacity: 0.5;
+  }
+  & .${sliderUnstyledClasses.rail} {
+    display: block;
+    position: absolute;
+    width: 100%;
+    height: 4px;
+    border-radius: 2px;
+    background-color: currentColor;
+    opacity: 0.4;
+  }
+  & .${sliderUnstyledClasses.track} {
+    display: block;
+    position: absolute;
+    height: 4px;
+    border-radius: 2px;
+    background-color: currentColor;
+  }
+  & .${sliderUnstyledClasses.thumb} {
+    position: absolute;
+    width: 16px;
+    height: 16px;
+    margin-left: -6px;
+    margin-top: -6px;
+    box-sizing: border-box;
+    border-radius: 50%;
+    outline: 0;
+    border: 3px solid currentColor;
+    background-color: #fff;
+    :hover,
+    &.${sliderUnstyledClasses.focusVisible} {
+      box-shadow: 0 0 0 0.25rem ${alpha(
+        theme.palette.mode === 'light' ? blue[400] : blue[300],
+        0.15,
+      )};
+    }
+    & .label {
+        background: unset;
+        background-color: ${theme.palette.mode === 'light' ? blue[500] : blue[300]};
+        width: 32px;
+        height: 32px;
+        padding: 0px;
+        visibility: hidden;
+        color: #fff;
+        border-radius: 50% 50% 50% 0;
+        position: absolute;
+        transform: translate(-35%, -140%) rotate(-45deg) scale(0);
+        transition: transform 0.3s ease;
+    }
+    :hover .label {
+        visibility: visible;
+        transform: translate(-35%, -140%) rotate(-45deg) scale(1);
+    }
+    :hover .value {
+        transform: rotate(45deg);
+        text-align: center;
+        line-height: 1.8
+    }
+    &.${sliderUnstyledClasses.active} {
+      box-shadow: 0 0 0 0.25rem ${alpha(
+        theme.palette.mode === 'light' ? blue[200] : blue[300],
+        0.3,
+      )};
+    }
+  }
+`,
+);
+
+interface SliderValueLabelProps {
+  children: React.ReactElement;
+}
+
+function SliderValueLabel({ children }: SliderValueLabelProps) {
+  return (
+    <span className="label">
+      <div className="value">{children}</div>
+    </span>
+  );
+}
+
+export default function UnstyledSlider() {
```

Comment:
```suggestion
export default function LabeledValuesSlider() {
```

and also rename the files.

Reply:
Another idea would be showing how the labels for the values can be shown during dragging (if the mouse is not over the thumb). This is how it works in Material UI - https://mui.com/material-ui/react-slider/#discrete-sliders